rfd/0113-moderated-file-transfers.md

```diff
@@ -0,0 +1,176 @@
+
+---
+authors: Michael Myers (michael.myers@goteleport.com)
+state: draft
+---
+
+# RFD 113 - Web Moderated Session File Transfer
+
+## Required approvers
+
+Engineering: @zmb3 @xacrimon @jakule
+
+## What
+
+Approval process to allow file transfers during a web moderated session
+
+## Why
+
+Users have no ability to transfer files if their role requires a moderated session for the specified server, even if they are in an active and "valid" session currently. Originally, we would allow file transfers regardless of moderated sessions which, while good for UX, goes against the purpose of moderated sessions in the first place. We have disabled the functionality but would like to reintroduce it in a secure, moderated, and auditable way. 
+
+## Scope
+
+This RFD will cover how to implement for web SCP specifically, but may be expanded to SFTP for tsh, although my research has been mostly for web scp. If at anytime there is ambiguity in this RFD for "which type of file transfers does this refer to", default to web scp only. 
```

Comment:
Are we leaving tsh out of scope because it requires a different solution or because we just haven't thought about it yet?

From a design perspective, I think it's important to understand why we're proposing a solution that's only valid for web.

Reply:
I believe the backend portion of this solution could be reused for `tsh` quite easily. I don't have the context yet to decide what the UI/UX would look like for `tsh`. I can do more research and revisit.

Reply:
Commenting a few more notes/thinking out loud. 

With the web, the request is outside of the shell and handled via the UI, as well as the approval process. This means the shell isn't blocked by any file transfer requests. 

For cli, we can maybe inform the approvers in the cli that someone is requesting a file transfer and implement an approve/deny command, similar to the force-terminate command. I'm not sure how this would scale will multiple file transfer requests at once, but its a thread to pull on.

I think the `tsh scp` command could function after approval similar to the new file transfer http request optional params by adding two new flags, the `--sessionId` and `--commandRequestId`.

 I'll continue to ponder.